logspout/logspout.go

```diff
@@ -65,32 +70,42 @@ func syslogStreamer(target Target, types []string, logstream chan *Log) {
 			continue
 		}
 		tag, pid, data := getLogParts(logline)
-		var conn net.Conn
+
+		// HACK: Go's syslog package hardcodes the log format, so let's send our own message
+		data = fmt.Sprintf("%s %s[%s]: %s",
+			time.Now().Format(getopt("DATETIME_FORMAT", dtime.DeisDatetimeFormat)),
+			tag,
+			pid,
+			data)
+
 		if strings.EqualFold(target.Protocol, "tcp") {
 			addr, err := net.ResolveTCPAddr("tcp", target.Addr)
 			assert(err, "syslog")
-			tcpconn, err := net.DialTCP("tcp", nil, addr)
+			conn, err := net.DialTCP("tcp", nil, addr)
+			assert(err, "syslog")
+			assert(conn.SetWriteBuffer(MAX_TCP_MSG_BYTES), "syslog")
+			_, err = fmt.Fprintln(conn, data)
 			assert(err, "syslog")
-			assert(tcpconn.SetWriteBuffer(1048576), "syslog")
-			conn = tcpconn
 		} else if strings.EqualFold(target.Protocol, "udp") {
+			// Truncate the message if it's too long to fit in a single UDP packet.
+			// Get the bytes first.  If the string has non-UTF8 chars, the number of
+			// bytes might exceed the number of characters and it would be good to
+			// know that up front.
+			dataBytes := []byte(data)
+			if len(dataBytes) > MAX_UDP_MSG_BYTES {
+				// Truncate the bytes and add ellipses.
+				dataBytes = append(dataBytes[:MAX_UDP_MSG_BYTES-3], "..."...)
```

Comment:
I don't think you need the second `...`. Here's an example: http://play.golang.org/p/98lLjaz5Vf